Sources/PostgresNIO/New/PSQLChannelHandler.swift

```diff
@@ -491,6 +463,19 @@ final class PSQLChannelHandler: ChannelDuplexHandler {
     }
 }
 
+extension PSQLChannelHandler: PSQLRowsDataSource {
+    
+    func request() {
+        let action = self.state.requestQueryRows()
+        self.run(action, with: self.handlerContext!)
```

Comment:
@weissi the stored [`ChannelHandlerContext`](https://github.com/vapor/postgres-nio/pull/153/files#r624473616) is used here. If we don't want to store the `ChannelHandlerContext` here, we could trigger the requestQueryRows by sending a `ChannelOutboundEvent`. What is the preferred way?

Reply:
It’s fine to store a context on a handler. The usual warnings that they’re not thread-safe and everybody who has your context has the same access to the pipeline as the handler has. But here you’re storing it on the handler which is totally fine